service/timestreamwrite/api_client.go

```diff
@@ -24,6 +29,8 @@ const ServiceAPIVersion = "2018-11-01"
 // Write.
 type Client struct {
 	options Options
+
+	endpointCache *discovery.EndpointCache
 }
```

Comment:
it doesn't look like there is any way to configure endpoint discovery from the `aws.Config` value. I'm not seeing where the `LoadDefaultConfig` getting the endpoint discovery flag from the env or shared config.


The LoadDefaultConfig probably needs to be updated to populate a `EndpointDiscovery` field on the `aws.Config` structure. And define the enum in the `aws` or similar package.